var/spack/repos/builtin/packages/qt-declarative/package.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class QtDeclarative(CMakePackage):
+    """Qt Declarative (Quick 2)."""
+
+    homepage = "https://www.qt.io"
+    url = "https://github.com/qt/qtdeclarative/archive/refs/tags/v6.2.3.tar.gz"
+    list_url = "https://github.com/qt/qtdeclarative/tags"
+
+    maintainers = ["wdconinc", "sethrj"]
+
+    version("6.3.1", sha256="1606723c2cc150c9b7339fd33ca5e2ca00d6e738e119c52a1d37ca12d3329ba9")
+    version("6.3.0", sha256="b7316d6c195fdc31ecbf5ca2acf2888737539919a02ff8f11a911432d50c17ac")
+    version("6.2.4", sha256="cd939d99c37e7723268804b9516e32f8dd64b985d847469c78b66b5f4481c548")
+    version("6.2.3", sha256="eda82abfe685a6ab5664e4268954622ccd05cc9ec8fb16eaa453c54900591baf")
+
+    generator = "Ninja"
+
+    # Changing default to Release for typical use in HPC contexts
+    variant(
+        "build_type",
+        default="Release",
+        values=("Release", "Debug", "RelWithDebInfo", "MinSizeRel"),
+        description="CMake build type",
+    )
+
+    depends_on("cmake@3.16:", type="build")
+    depends_on("ninja", type="build")
+    depends_on("pkgconfig", type="build")
+    depends_on("python", when="@5.7.0:", type="build")
+
+    _versions = ["6.3.1", "6.3.0", "6.2.4", "6.2.3"]
+    for v in _versions:
+        depends_on("qt-base@" + v, when="@" + v)
+        depends_on("qt-shadertools@" + v, when="@" + v)
```

Comment:
This anti-pattern of storing the version list twice seems like it should be possible to avoid...

Reply:
See https://github.com/spack/spack/pull/14002 for some preliminary work to make this simpler.

Reply:
What is the timeline on that pr getting finalized?

Reply:
No timeline ☹️